### PR TITLE
Adding permission definition based on YML files.

### PR DESCRIPTION
**Pull request by RoySegall**
#44

@amitaibu @damiankloip What do you think?

### CODE REVIEW

File: tests/src/Kernel/Entity/OgPermissionIntegration.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\og\Og;
+
+/**
+ * Test OG permission mechanism.
+ *
+ * @group og
+ */
+class OgPermissionIntegration extends KernelTestBase {
```
Comment (amitaibu) [inline tests/src/Kernel/Entity/OgPermissionIntegration.php]: Can we `UnitTest`?

File: og.og_permissions.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+update group:
+  title: 'Edit group'
+  description: 'Edit the group. Note: This permission controls only node entity type groups.'
+  'default role':
+    - OG_ADMINISTRATOR_ROLE
+administer group:
+  title: 'Administer group'
+  description: 'Manage group members and content in the group.'
+  'restrict access': TRUE
+  'default role':
```
Comment (amitaibu) [inline og.og_permissions.yml:10]: Is this an optional property (as in OG7)?

Comment (amitaibu) [inline og.og_permissions.yml:10]: Also there's the 'roles' array (see og_ui_og_permission)

**Comment by RoySegall** [comment]
@amitaibu This one ready for merge.

File: src/OgPermissionHandler.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ *   # (optional) Additional description fo the permission used in the UI.
+ *   description: 'Edit the group. Note: This permission controls only node entity type groups.'
+ *   # (optional) Boolean, when set to true a warning about site security will
+ *   # be displayed on the Permissions page. Defaults to false.
+ *   restrict access: false
+ *   # Determine to which roles the permissions will be enabled by default.
+ *   'default role':
+ *     - OG_ADMINISTRATOR_ROLE
+ *   # Determine to which role to limit the permission. For example the
+ *   # "subscribe" can't be assigned only to a non-member, as a member doesn't
```
Comment (amitaibu) [inline src/OgPermissionHandler.php]: can't =>  can

File: src/OgPermissionHandler.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function buildPermissionsYaml() {
+    $permissions = parent::buildPermissionsYaml();
+
+    foreach ($permissions as &$permission) {
+      $permission += [
```
Comment (amitaibu) [inline src/OgPermissionHandler.php:60]: `// Add default values.`

File: tests/src/Unit/OgPermissionHandlerTest.php [hunk 5]
```diff
@@ -0,0 +1,214 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\og\Unit\OgPermissionHandlerTest.
+ */
```
Comment (amitaibu) [inline tests/src/Unit/OgPermissionHandlerTest.php:6]: I would add a reference to the core class it was copy/ pasted and adjusted from.

File: src/OgPermissionHandler.php (hunk 3 above)
Comment (RoySegall) [inline src/OgPermissionHandler.php]: Migrated from og but I'll fix it.

**Comment by RoySegall** [comment]
@amitaibu I pushed the changes.

**Comment by amitaibu** [comment]
Thanks.  We need another pr to port modules permissions to the new yml format.

**Pull request merged**